src/js/profile/mobile/widget/DatePicker.js

```diff
@@ -0,0 +1,416 @@
+* @since 1.2
+* @component-selector .ui-date-picker
+* @extends ns.widget.mobile.DatePicker
+* @author Hunseop Jeong <hs85.jeong@samsung.com>
+*/
+(function (window, document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../../core/engine",
+			"../../../core/event",
+			"../../../core/event/gesture",
+			"../../../core/util/selectors",
+			"./BaseWidgetMobile",
+			"./Spin"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var BaseWidget = ns.widget.mobile.BaseWidgetMobile,
+				engine = ns.engine,
+				utilsEvents = ns.event,
+				getClosestByClass = ns.util.selectors.getClosestByClass,
+				Spin = ns.widget.mobile.Spin,
+				WIDGET_CLASS = "ui-date-picker",
+				classes = {
+					HEADER: WIDGET_CLASS + "-header",
+					CONTENT: WIDGET_CLASS + "-content",
+					CONTAINER: WIDGET_CLASS + "-container",
+					DAY_CONTAINER: WIDGET_CLASS + "-container-day",
+					MONTH_CONTAINER: WIDGET_CLASS + "-container-month",
+					YEAR_CONTAINER: WIDGET_CLASS + "-container-year"
+				},
+				MONTH_NAMES = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
+				MIN_YEAR = 1900,
+				MAX_YEAR = 2050,
+				WIDGET_SELECTOR = "." + WIDGET_CLASS,
+				DatePicker = function () {
+					this.options = {
+						view: "wheel"
+					};
+
+					this._spins = {
+						day: null,
+						month: null,
+						year: null
+					};
+					this._ui = {
+						header: null
+					};
+				},
+				prototype = new BaseWidget();
+
+			DatePicker.classes = classes;
+
+			/**
+			* Init method
+			* @method _init
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._init = function () {
+				var self = this;
+
+				self._setValue(new Date());
+				// Update day in month
+				self._changeMonth(self._getValue("month"));
+
+				// Calculate the spin widget after building the widget
+				Object.keys(self._spins).forEach(function (key) {
+					self._spins[key].refresh();
+				});
+			};
+
+			/**
+			* Build widget for the date picker
+			* @method _buildDatePicker
+			* @param {string} name
+			* @return {HTMLElement} container for date spin
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._buildDatePicker = function (name) {
+				var self = this,
+					ui = self._ui,
+					spin = document.createElement("div"),
+					spinContainer = document.createElement("div"),
+					options = {},
+					spinWidget;
+
+				spin.classList.add(Spin.classes.SPIN);
+				spinContainer.classList.add(DatePicker.classes.CONTAINER);
+
+				if (name === "day") {
+					options = {
+						min: 1,
+						max: 31
+					};
+					spinContainer.classList.add(DatePicker.classes.DAY_CONTAINER);
+				} else if (name === "month") {
+					options = {
+						min: 1,
+						max: MONTH_NAMES.length,
+						labels: MONTH_NAMES.map(function (value) {
+							return value.toUpperCase().substring(0, 3);
```

Comment:
Why is substring (0,3)?
What is the value value? And I am wondering why .join does it.